airflow/contrib/hooks/sftp_hook.py

```diff
@@ -39,32 +42,66 @@ class SFTPHook(BaseHook):
     Errors that may occur throughout but should be handled downstream.
     """
 
-    def __init__(self, ftp_conn_id='sftp_default'):
-        self.ftp_conn_id = ftp_conn_id
+    def __init__(self, ftp_conn_id='sftp_default', *args, **kwargs):
+        kwargs['ssh_conn_id'] = ftp_conn_id
+        super(SFTPHook, self).__init__(*args, **kwargs)
+
         self.conn = None
+        self.private_key_pass = None
+
+        if self.ssh_conn_id is not None:
+            conn = self.get_connection(self.ssh_conn_id)
+            if conn.extra is not None:
+                extra_options = conn.extra_dejson
+                if 'private_key_pass' in extra_options:
+                    self.private_key_pass = extra_options.get('private_key_pass', None)
+
+                # For backward compatibility
+                # TODO: remove in Airflow 2.1
+                import warnings
+                if 'ignore_hostkey_verification' in extra_options \
+                        and str(extra_options["ignore_hostkey_verification"])\
+                        .lower() == 'false':
```

Comment:
I think this line should check for `true` instead, because the `ignore_hostkey_verification` option was the inverse of the `no_host_key_check` option. If `ignore_hostkey_verification` was `true`, no check could be performed.

Reply:
Oops you are right. @ckljohn or @Gwildor could one of you open a small PR (using the same JIRA) to fix this?

Reply:
I'll have a look, thanks!

Reply:
@ashb my PR is #4085

Reply:
Thanks.